core/src/main/java/bio/terra/pearl/core/service/notification/NotificationService.java

```diff
@@ -56,6 +56,10 @@ public List<Notification> findAllByConfigId(UUID configId, boolean withEnrollees
             attachEnrollees(notifications);
         }
 
+        for (Notification notification : notifications) {
+            dao.attachSendgridEvent(notification);
```

Comment:
this is going to be 1000+ separate DB calls for most production configs.  Let's make a new `attachAllSendgridEvents(List<Notification> notifications)` dao method that grabs all the events in one query